frontend/src/routes/create-goals/+page.svelte

```diff
@@ -49,170 +49,275 @@
+              bind:value={goal.time_bound}
+            />
+            <div class="mt-3">
+              <button class="btn btn-primary">Keep Time-Bound Suggestion</button>
+            </div>
+          </figcaption>
+        </figure>
+      </div>
+    </div>
+    <div class="card">
       <figure>
         <figcaption class="p-4 card-body">
-          <h2 class="card-title">Relevant</h2>
-          <input
-            id="relevant"
-            class="shadow appearance-none border rounded w-full py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
-            type="text"
-            placeholder="AchievAIm's Relevant suggestion"
-            bind:value={goal.relevant}
-          />
-          <div class="mt-3">
-            <button class="btn btn-primary">Keep Relevant Suggestion</button>
+          <div class="mt-4 pd-4 flex flex-col items-center">
+            {#if goal.days_of_week}
+              <span class="block text-xl font-bold mb-2"
+                >Select the Days Your SMART Goal Repeats:</span
+              >
+              <button class="btn btn-primary mt-4" on:click={toggleAll}>
+                {#if selectAll}Deselect All{:else}Select All{/if}
+              </button>
+              <DaysOfWeekSelector daysOfWeek={goal.days_of_week} />
+            {/if}
           </div>
-        </figcaption>
-      </figure>
-    </div>
+          <div class="card">
+            <figure>
+              <figcaption class="p-4 card-body">
+                <label class="block text-lg font-bold mb-2 mt-4" for="goal-time">
+                  Set the alert time for your SMART goals on selected days.
+                </label>
+                <input
+                  class="shadow m-2 appearance-none border rounded w-52 py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
+                  id="goal-time"
+                  type="time"
+                  bind:value={goal.time_of_day}
+                />
+              </figcaption>
+            </figure>
+          </div>
+          <div class="card">
+            <figure>
+              <figcaption class="p-4 card-body">
+                <label class="block text-lg font-bold mb-2" for="goal-date">
+                  Choose the Date for Completing Your SMART Goal.
+                </label>
+                <input
+                  class="shadow w-52 appearance-none border rounded py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
+                  id="goal-date"
+                  type="date"
+                  bind:value={goal.date_for_achievement}
+                />
+              </figcaption>
+            </figure>
+          </div>
+          <div class="card">
+            <figure>
+              <figcaption class="p-4 card-body">
+                <div class="mt-4 flex flex-col items-center">
+                  <button class="btn btn-primary" on:click={handleSave}>Save Smart Goal</button>
```

Comment:
I'm getting errors when I try to save. I haven't had a chance to look into why yet. The error message implies that either all of the values are not being sent, or some of the values are bad. My guess is that the error is coming from the date and not related to this PR?

Reply:
Are you talking about the above ↑ https://github.com/xyvora/achievaim/pull/254/files/2632dd26c8f80a1fb22f1625203fc8c5b2a8dcd9#r1279959293?